teuthology/task/kernel.py

```diff
@@ -188,10 +188,22 @@ def need_to_install(ctx, role, version):
                                                                  want=version))
 
     if '.' in str(version):
-        # version is utsrelease, yay
         if cur_version == version:
             log.debug('utsrelease strings match, do not need to install')
             ret = False
+        os_type = teuthology.get_distro(ctx)
+        log.debug("Distro of this test job: {}".format(os_type))
+        if os_type in ['sle', 'opensuse']:
+            cur_version_match = re.search('(.*)-default$', cur_version)
+            if cur_version_match:
+                cur_version_rp = cur_version_match.group(1)
+                if cur_version_rp in version:
+                    log.debug('"{}" is a substring of "{}" - the latest {} kernel is running'
```

Comment:
Since we are not going to support _luminous_ maybe we can use _f_ strings instead of .format, like:
```
log.debug(f"'{cur_versioin_rp}' is a substring of '{version}' - the latest {os_type} kernel is running")
```